platform/security/sts/security-sts-propertyclaimshandler/src/main/java/org/codice/ddf/security/sts/claims/property/AttributeFileClaimsHandler.java

```diff
@@ -37,57 +44,32 @@
 import org.apache.cxf.sts.claims.ClaimsParameters;
 import org.apache.cxf.sts.claims.ProcessedClaim;
 import org.apache.cxf.sts.claims.ProcessedClaimCollection;
-import org.apache.cxf.sts.token.realm.RealmSupport;
 import org.boon.json.JsonFactory;
-import org.boon.json.ObjectMapper;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-public class AttributeFileClaimsHandler implements ClaimsHandler, RealmSupport {
-
-  private static final Logger LOGGER = LoggerFactory.getLogger(AttributeFileClaimsHandler.class);
+import org.codice.ddf.configuration.SystemBaseUrl;
 
-  private static final ObjectMapper MAPPER = JsonFactory.create();
+public class AttributeFileClaimsHandler implements ClaimsHandler, SystemHighAttributes {
 
   private String attributeFileLocation;
 
-  private List<URI> supportedClaimTypes = new ArrayList<>();
+  private List<URI> supportedClaimTypes;
 
   private Map<String, Object> json;
 
-  private List<String> supportedRealms;
+  private Map<String, List<Serializable>> systemHighAttributes;
 
-  private String realm;
-
-  public void init() {
-    if (attributeFileLocation != null) {
-      Path ddfHomePath = Paths.get(System.getProperty("ddf.home"));
-      Path path = Paths.get(attributeFileLocation);
-      if (!path.isAbsolute()) {
-        path = Paths.get(ddfHomePath.toString(), path.toString());
-      }
-      Set<String> claims = new HashSet<>();
-      try (InputStream stream = Files.newInputStream(path)) {
-        String jsonString = IOUtils.toString(stream);
-        json = MAPPER.parser().parseMap(jsonString);
-        Set<Map.Entry<String, Object>> entries = json.entrySet();
-        for (Map.Entry<String, Object> entry : entries) {
-          Object value = entry.getValue();
-          if (value instanceof Map) {
-            Set keySet = ((Map) value).keySet();
-            for (Object key : keySet) {
-              claims.add((String) key);
-            }
-          }
-        }
-        supportedClaimTypes.clear();
-        supportedClaimTypes.addAll(claims.stream().map(URI::create).collect(Collectors.toList()));
-      } catch (IOException e) {
-        LOGGER.info(
-            "Unable to read attribute file for system users. Set log level for \"org.codice.ddf.security.sts.claims.property\" to DEBUG for more information.");
-        LOGGER.debug("Unable to read attribute file for system users.", e);
-      }
+  public void init() throws IOException {
+    Path ddfHomePath = Paths.get(System.getProperty("ddf.home"));
+    Path path = Paths.get(attributeFileLocation);
+    if (!path.isAbsolute()) {
+      path = Paths.get(ddfHomePath.toString(), path.toString());
+    }
+    try (InputStream stream = Files.newInputStream(path)) {
+      String jsonString = IOUtils.toString(stream, Charset.defaultCharset());
+      json = JsonFactory.create().parser().parseMap(jsonString);
     }
```

Comment:
❓ Shouldn't we keep logging something like we used to (L85-89) when the file can't be read or parsed?

Reply:
I removed these log statements because I'd like the initialization of `AttributeFileClaimsHandler` to fail (i.e. throw and exception) if there is an issue reading the file. I don't want to catch the exception and report that the `AttributeFileClaimsHandler` was successfully initialized.